examples/step-51/step-51.cc

```diff
@@ -268,48 +247,43 @@ namespace Step51
       : Function<dim>()
     {}
 
-    virtual double value(const Point<dim> & p,
-                         const unsigned int component = 0) const override;
-
-  private:
-    const ConvectionVelocity<dim> convection_velocity;
-  };
+    virtual double value(const Point<dim> &p,
+                         const unsigned int /*component*/ = 0) const override
+    {
+      ConvectionVelocity<dim> convection_velocity;
+      Tensor<1, dim>          convection   = convection_velocity.value(p);
+      double                  return_value = 0;
+      for (unsigned int i = 0; i < this->n_source_centers; ++i)
+        {
+          const Tensor<1, dim> x_minus_xi = p - this->source_centers[i];
 
+          return_value +=
+            ((2 * dim - 2 * convection * x_minus_xi -
+              4 * x_minus_xi.norm_square() / (this->width * this->width)) /
+             (this->width * this->width) *
+             std::exp(-x_minus_xi.norm_square() / (this->width * this->width)));
+        }
 
-  template <int dim>
-  double RightHandSide<dim>::value(const Point<dim> &p,
-                                   const unsigned int) const
-  {
-    Tensor<1, dim> convection   = convection_velocity.value(p);
-    double         return_value = 0;
-    for (unsigned int i = 0; i < this->n_source_centers; ++i)
-      {
-        const Tensor<1, dim> x_minus_xi = p - this->source_centers[i];
+      return return_value / Utilities::fixed_power<dim>(
+                              std::sqrt(2 * numbers::PI) * this->width);
```

Comment:
Same here

In both cases, the name of the variable is wrong because that's not the value being returned

Reply:
Good observation, let's call the variable `sum`.